File: othermodule/src/main/java/me/yifeiyuan/ktx/foundation/othermodule/vb/ViewBindingComponent.java
package me.yifeiyuan.ktx.foundation.othermodule.vb;

import android.view.View;

import androidx.annotation.NonNull;
import androidx.viewbinding.ViewBinding;

import me.yifeiyuan.flap.Component;
import me.yifeiyuan.flap.annotations.Proxy;
import me.yifeiyuan.ktx.foundation.othermodule.databinding.FlapItemVbBinding;

/**
 * Created by 程序亦非猿 on 2020/9/30.
 */
@Proxy(layoutName = "flap_item_vb", useViewBinding = true)
public class ViewBindingComponent extends Component<VBModel> {

     FlapItemVbBinding binding;
    public ViewBindingComponent(@NonNull FlapItemVbBinding binding) {
        super(binding.getRoot());
        this.binding = binding;
    }

    @Override
    protected void onBind(@NonNull VBModel model) {
        binding.tvContent.setText("ViewBinding Sample");
    }
}

File: othermodule/src/main/java/me/yifeiyuan/ktx/foundation/othermodule/vb/ViewBindingComponent.kt
package me.yifeiyuan.ktx.foundation.othermodule.vb

import android.view.LayoutInflater
import android.view.ViewGroup
import me.yifeiyuan.flap.delegate.AdapterDelegate
import me.yifeiyuan.flap.Component
import me.yifeiyuan.flap.annotations.Delegate
import me.yifeiyuan.ktx.foundation.othermodule.databinding.FlapItemVbBinding

/**
 * 使用 ViewBinding 的例子
 * Created by 程序亦非猿 on 2020/9/30.
 */
@Delegate(layoutName = "flap_item_vb", useViewBinding = true)
class ViewBindingComponent(var binding: FlapItemVbBinding) : Component<ViewBindingModel>(binding.root) {

    override fun onBind(model: ViewBindingModel) {
        binding.tvContent.text = "ViewBinding Sample"
    }

}

class ViewBindingComponentDelegate : AdapterDelegate<ViewBindingModel, ViewBindingComponent> {
    override fun onCreateViewHolder(inflater: LayoutInflater, parent: ViewGroup, viewType: Int): ViewBindingComponent {
        return ViewBindingComponent(FlapItemVbBinding.inflate(inflater, parent, false))
    }

    override fun delegate(model: Any): Boolean {
        return ViewBindingModel::class.java == model::class.java
    }
}